eradiate/scenes/measure/_distant.py

```diff
@@ -593,3 +613,160 @@ def _postprocessing_add_reflectance(self, ds: xr.Dataset) -> xr.Dataset:
+    target a shape which can be controlled through the ``target`` parameter.
+    This feature is useful if one wants to compute the average flux leaving
+    a particular subset of the scene.
+
+    .. admonition:: Notes
+       :class: note
+
+       * Setting the ``target`` parameter is required to get meaningful results.
+         Solver applications should take care of setting it appropriately.
+       * The film resolution can be adjusted to manually stratify film sampling
+         and reduce variance in results. The default 32x32 is generally a good
+         choice, but scenes with sharp reflection lobes may benefit from higher
+         values.
+       * This scene element is a thin wrapper around the ``distantflux``
+         sensor kernel plugin.
+    """
+
+    direction = documented(
+        attr.ib(
+            default=[0, 0, 1],
+            converter=np.array,
+            validator=validators.is_vector3,
+        ),
+        doc="A 3-vector defining the normal to the reference surface for which "
+        "the exitant flux density is computed.",
+        type="array-like",
+        default="[0, 0, 1]",
+    )
+
+    _film_resolution = documented(
+        attr.ib(
+            default=(32, 32),
+            validator=attr.validators.deep_iterable(
+                member_validator=attr.validators.instance_of(int),
+                iterable_validator=validators.has_len(2),
+            ),
+        ),
+        doc="Film resolution as a (width, height) 2-tuple.",
+        type="array-like",
+        default="(32, 32)",
+    )
+
+    @property
+    def film_resolution(self) -> Tuple[int, int]:
+        return self._film_resolution
+
+    def _base_dicts(self) -> List[Dict]:
+        from mitsuba.core import ScalarTransform4f, ScalarVector3f, coordinate_system
+
+        result = []
+        _, up = coordinate_system(self.direction)
+
+        for sensor_info in self.sensor_infos():
+            d = {
+                "type": "distantflux",
+                "id": sensor_info.id,
+                "to_world": ScalarTransform4f.look_at(
+                    origin=[0, 0, 0],
+                    target=ScalarVector3f(self.direction),
+                    up=up,
+                ),
+            }
+
+            if self.target is not None:
+                d["target"] = self.target.kernel_item()
+
+            if self.origin is not None:
+                d["origin"] = self.origin.kernel_item()
+
+            result.append(d)
+
+        return result
+
+    def postprocess(self) -> xr.Dataset:
+        # Fetch results (SPP-split aggregated) as a Dataset
+        result = self._postprocess_fetch_results()
+
+        return result
+
+    def _postprocess_fetch_results(self) -> xr.Dataset:
+        # Collect results and add appropriate metadata
+        ds = self.results.to_dataset(aggregate_spps=True)
+
+        # Add aggregate flux density field
+        ds["flux"] = ds["raw"].mean(dim=("x", "y"))
```

Comment:
That's indeed unintuitive: each pixel stores the flux as if it would be integrated over the entire hemisphere (see the [`distantflux` plugin docs](https://eradiate-kernel.readthedocs.io/en/latest/generated/plugins.html#distant-fluxmeter-sensor-distantflux)). I'll see if I can get the flux density for a subset of the hemisphere with a PDF factor.

Reply:
I guess you'll like this better :)